server/cmd/dcrdex/main.go

```diff
@@ -0,0 +1,86 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/decred/dcrdex/server/asset"
+	_ "github.com/decred/dcrdex/server/asset/btc" // register btc asset
+	_ "github.com/decred/dcrdex/server/asset/dcr" // register dcr asset
+	_ "github.com/decred/dcrdex/server/asset/ltc" // register ltc asset
+	"github.com/decred/slog"
+)
+
+func mainCore() error {
+	// Parse the configuration file, and setup logger.
+	cfg, err := loadConfig()
+	if err != nil {
+		fmt.Printf("Failed to load dcrdata config: %s\n", err.Error())
+		return err
+	}
+	defer func() {
+		if logRotator != nil {
+			logRotator.Close()
+		}
+	}()
+
+	// Display app version.
+	log.Infof("%s version %v (Go version %s)", AppName, Version(), runtime.Version())
+	log.Infof("dcrdex starting for network: %s", cfg.Network)
+
+	// Load the market and asset configurations for the given network.
+	markets, assets, err := loadMarketConf(cfg.Network)
+	if err != nil {
+		return fmt.Errorf("failed to load market and asset config: %v", err)
+	}
+	log.Infof("Found %d assets, loaded %d markets, for network %s",
+		len(assets), len(markets), strings.ToUpper(cfg.Network.String()))
+
+	ctx, shutdown := context.WithCancel(context.Background())
+	defer func() {
+		shutdown()
+		time.Sleep(time.Second)
+	}()
+
+	// Start asset backends.
+	backends := make(map[string]asset.DEXAsset, len(assets))
+	for _, asseti := range assets {
+		name := asseti.Bip44Symbol
+		subsys := fmt.Sprintf("ASSET[%s]", name)
+		logger := slog.NewBackend(os.Stdout).Logger(subsys)
+		log.Infof("Starting asset backend %q...", name)
+		be, err := asset.Start(name, ctx, asseti.ConfigPath, logger, cfg.Network)
+		if err != nil {
+			return fmt.Errorf("failed to start asset %q: %v", name, err)
+		}
+		backends[name] = be
+	}
```

Comment:
Here is where the assets driver system helps.  Each asset packages registers by the asset's BIP44 symbol, allowing an asset to be created by `asset.Start(name, ...`)